pareto/utilities/bounding_functions.py

```diff
@@ -0,0 +1,161 @@
+#####################################################################################################
+# PARETO was produced under the DOE Produced Water Application for Beneficial Reuse Environmental
+# Impact and Treatment Optimization (PARETO), and is copyright (c) 2021-2023 by the software owners:
+# The Regents of the University of California, through Lawrence Berkeley National Laboratory, et al.
+# All rights reserved.
+#
+# NOTICE. This Software was developed under funding from the U.S. Department of Energy and the U.S.
+# Government consequently retains certain rights. As such, the U.S. Government has been granted for
+# itself and others acting on its behalf a paid-up, nonexclusive, irrevocable, worldwide license in
+# the Software to reproduce, distribute copies to the public, prepare derivative works, and perform
+# publicly and display publicly, and to permit others to do so.
+#####################################################################################################
+# Title: Supporting functions for both Operational and Strategic models
+
+###############################################################################
+### Imports
+from pareto.strategic_water_management.strategic_produced_water_optimization import (
+    WaterQuality,
+)
+from pyomo.environ import (
+    Param,
+    value,
+    Any,
+)
+
+
+###############################################################################
+### Variable bounding manager functions
+def VariableBounds(model):
+    model = Bound_v_F_Piped(model)
+    if model.config.water_quality is WaterQuality.post_process:
+        model = Bound_v_Q_PP(model)
+    elif model.config.water_quality is WaterQuality.discrete:
+        model = Bound_v_F_DiscretePiped(model)
+        # model = Bound_v_F_Trucked(model) # TODO: resolve unit problem with p_max_number_of_trucks - it must be week^-1 to make the units work out
```

Comment:
Noticed this TODO. Is it still something that is needed?

Reply:
This will require a change in the corresponding PARETO parameter in the strategic model code. I built this in for future reference. At this point, I would rather keep this as-is and raise this in a future update.

Reply:
The parameter p_max_number_of_trucks will need to be available within the primary model block, or else this bound can only be added to specific subtypes. Moreover, the units on the parameter are [unitless] but I have a strong argument for making the units for p_max_number_of_trucks [week^-1]. I think this will need to be a follow-up PR once this first version is merged.